src/GenAPI/Microsoft.DotNet.GenAPI/SyntaxGeneratorExtensions.cs

```diff
@@ -33,6 +39,21 @@ public static SyntaxNode DeclarationExt(this SyntaxGenerator syntaxGenerator, IS
                 }
             }
 
+            if (symbol.Kind == SymbolKind.Method)
+            {
+                IMethodSymbol method = (IMethodSymbol)symbol;
+                if (method.MethodKind == MethodKind.Constructor)
+                {
+                    INamedTypeSymbol? baseType = method.ContainingType.BaseType;
+                    // If the base type does not have default constructor.
+                    if (baseType != null && !baseType.Constructors.IsEmpty && baseType.Constructors.Where(c => c.Parameters.IsEmpty).Count() == 0)
```

Comment:
When using LINQ, we usually prefer `!Any()` over `Count() == 0` but in this case, an All operator might even be more readable.

```suggestion
                    if (baseType != null && !baseType.Constructors.IsEmpty && baseType.Constructors.All(c => !c.Parameters.IsEmpty))
```